internal/model/modelindexer/indexer.go

```diff
@@ -384,15 +442,33 @@ func (i *Indexer) flush(ctx context.Context, bulkIndexer *bulkIndexer) error {
 // outgoing Elasticsearch bulk requests are flushed due to the idle timer,
 // rather than due to being full.
 func (i *Indexer) runActiveIndexer() {
-	atomic.AddInt64(&i.activeBulkRequests, 1)
-	defer atomic.AddInt64(&i.activeBulkRequests, -1)
 	var closed bool
 	var active *bulkIndexer
+	var timedFlush uint
+	var fullFlush uint
 	flushTimer := time.NewTimer(i.config.FlushInterval)
+	idleTimer := time.NewTimer(i.config.Scaling.IdleInterval)
 	if !flushTimer.Stop() {
 		<-flushTimer.C
 	}
-	handleBulkItem := func(event elasticsearch.BulkIndexerItem) (flush bool) {
+	if !idleTimer.Stop() {
+		<-idleTimer.C
+	}
+	countTimedFlush := func() {
+		if i.config.Scaling.Disabled {
+			return
+		}
+		timedFlush++
+		fullFlush = 0
+	}
+	countFullFlush := func() {
+		if i.config.Scaling.Disabled {
+			return
+		}
+		fullFlush++
+		timedFlush = 0
+	}
```

Comment:
It's probably faster to just increment/clear the counters inline. This code might get inlined, but even if it is there's some branching in there (i.e. the conditional on Disabled). I don't think it's harmful to update the counters even if scaling is disabled, is it?